django/db/migrations/operations/models.py

```diff
@@ -483,6 +483,28 @@ def reduce(self, operation, app_label=None):
         return super().reduce(operation, app_label=app_label)
 
 
+class AlterModelBases(ModelOperation):
+    reduce_to_sql = False
+    reversible = True
+
+    def __init__(self, model_name, bases):
+        self.model_name = model_name
+        self.bases = bases
+
+    def state_forwards(self, app_label, state):
+        state.models[app_label, self.model_name].bases = self.bases
+        state.reload_model(app_label, self.model_name)
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        pass
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        pass
+
+    def describe(self):
+        return "Update %s bases to %s" % (self.model_name, self.bases)
```

Comment:
I would rename `self.model_name` -> `self.name` (like in `CreateModel`, `DeleteModel`, or `RenameModel`).